### PR TITLE
Added pallet for risc0 verifier

### PR DESCRIPTION
**Pull request by JackPiri**

### CODE REVIEW

File: pallets/settlement-risc0/src/benchmarking.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    submit_proof {
+        let caller: T::AccountId = whitelisted_caller();
+    }: _(RawOrigin::Signed(caller), VALID_VK.into(), VALID_PROOF.to_vec().into(), VALID_PUBS.to_vec().into())
+}
+
+impl_benchmark_test_suite!(
+    SettlementRisc0Pallet,
+    crate::mock::new_test_ext(),
+    crate::mock::Test,
+);
```
Comment (la10736) [inline pallets/settlement-risc0/src/benchmarking.rs:36]: That's quite semplicistic. This benchmark doesn't take care that the verify time is not costants but depends by program's cycle number that shoulb reflected in the proof size.

@DanieleDiBenedetto For now we can take this like it is bu maybe is better to add a ticket to handle it later.

File: pallets/settlement-risc0/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            proof: Box<Vec<u8>>,
+            pubs: Box<Vec<u8>>,
+        ) -> DispatchResultWithPostInfo {
+            log::trace!("Submitting proof");
+            verify_proof::<T>(vk_u8, *proof, *pubs)
+                .map(Into::into)
+                .map_err(Into::into)
+        }
+    }
+
```
Comment (la10736) [inline pallets/settlement-risc0/src/lib.rs:119]: You can remove all `Box` here: take care of remove also all into inthe related calls

**Review by la10736 — CHANGES_REQUESTED** [review]
Great work. Overall is OK... Just remove the unneeded `Box`'s in submit_proof extrinsic, add integration and E2E tests

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -340,6 +340,11 @@ impl pallet_settlement_zksync::Config for Runtime {
     type WeightInfo = weights::pallet_settlement_zksync::NHWeight<Runtime>;
 }
 
+impl pallet_settlement_risc0::Config for Runtime {
+    type OnProofVerified = Poe;
+    type WeightInfo = weights::pallet_settlement_risc0::NHWeight<Runtime>;
```
Comment (la10736) [inline runtime/src/lib.rs:345]: Maybe you neet to change it

File: runtime/src/tests.rs [hunk 4]
```diff
@@ -192,6 +192,26 @@ fn pallet_zksync_availability() {
...
+#[test]
+fn pallet_risc0_availability() {
+    new_test_ext().execute_with(|| {
+        let dummy_origin = AccountId32::new([0; 32]);
+
+        let dummy_vk = [0u8; 32];
+        let dummy_proof = vec![];
+        let dummy_pubs = vec![];
+
+        assert!(SettlementRisc0Pallet::submit_proof(
```
Comment (la10736) [inline runtime/src/tests.rs:204]: Add your test here

Comment (la10736) [reply runtime/src/tests.rs:204]: And also the availability one

File: runtime/src/weights.rs [hunk 5]
```diff
@@ -5,6 +5,7 @@ pub mod pallet_balances;
 pub mod pallet_im_online;
 pub mod pallet_poe;
```
Comment (la10736) [inline runtime/src/weights.rs:6]: You missed your module here

File: pallets/settlement-risc0/src/benchmarking.rs (hunk 1 above)
Comment (DanieleDiBenedetto) [reply pallets/settlement-risc0/src/benchmarking.rs]: Sure

**Review by DanieleDiBenedetto — APPROVED** [review]

File: runtime/src/lib.rs (hunk 3 above)
Comment (JackPiri) [reply runtime/src/lib.rs:345]: Yes, modified together with runtime tests.

File: runtime/src/tests.rs (hunk 4 above)
Comment (JackPiri) [reply runtime/src/tests.rs:204]: Added test for pallet availability and weight linking.

File: runtime/src/weights.rs (hunk 5 above)
Comment (JackPiri) [reply runtime/src/weights.rs:6]: Yes, added together with runtime tests.

**Review by la10736 — APPROVED** [review]

**Pull request merged**